README.md

```diff
@@ -37,14 +37,18 @@ auto res = encoder.encode(); // "~bidP~fsia@_cidP_gsia@_cidP_gsia@"
 encoder.clear(); // Clear the list of points.
 
 // Decode a string using static function.
-auto polyline = gepaf::PolylineEncoder<>::decode("~bidP~fsia@_cidP_gsia@_cidP_gsia@");
+auto polyline = gepaf2::PolylineEncoder<>::decode("~bidP~fsia@_cidP_gsia@_cidP_gsia@");
 
 // Iterate over all points and print coordinates of each.
 for (const auto &point : polyline) {
     printf("(%f, %f)\n", point.latitude(), point.longitude());
 }
 ```
 
+## Backward compatibility
+
+For backward compatibility reason with the previous version of library the `gepaf` namespace contains a `PolylineEncoder` class that is the very same as the `gepaf2::PolylineEncoder<5>`
```

Comment:
I really don't get the point here. Why do you rename the namespace if you even didn't change the library source code at all.

Reply:
i did change; in the polylineencoder.h i renamed the namespace and made a new namespace with the old hanem. this is what prevents the interface brake within the same major version.

Reply:
> i did change; in the polylineencoder.h i renamed the namespace and made a new namespace with the old hanem. this is what prevents the interface brake within the same major version.

But the question was **why did you rename the namespace**? You modified the build script (CMake), but what was wrong with the library code that forced you to rename the namespace?

Reply:
to make a gepaf namespace and add the 'PolylineEncoder' definition there. Without that you have to change the client code when you update from version 1.0.0 to 1.1.1 . Since the MAJOR version of the package didn't change, this interface break is not allowed (according to semver).

Reply:
I still don't get the idea. The namespace renaming seems to be an unnecessary trick to me. If my client code used the `gepaf` namespace before, it will still use it. I also don't understand the relation between CMake files changes and the namespace renaming.

Reply:
the idea is that without this block:  
```
namespace gepaf {
    using PolylineEncoder = gepaf2::PolylineEncoder<5>;
}
```
in the header file, you break the backward compatibility within the very same major release.  
  
consider the following: i use the version 1.0.0 of the polyline encoder lib. i use it as the following:  
```
const auto p = gepaf::PolylineEncoder();
```

i switch to version 1.1.1 and my code suddenly not compiling. To make it compile, i have to change the usage to  
```
const auto p = gepaf::PolylineEncoder<>();
```

The interface break is not allowed within the same major version, but it happened. Hence the new namespace i made and the old namespace with the using.

Reply:
the cmake change and the namespace change is not related.

Reply:
> the idea is that without this block:
> 
> ```
> namespace gepaf {
>     using PolylineEncoder = gepaf2::PolylineEncoder<5>;
> }
> ```
> 
> in the header file, you break the backward compatibility within the very same major release.
> 
> consider the following: i use the version 1.0.0 of the polyline encoder lib. i use it as the following:
> 
> ```
> const auto p = gepaf::PolylineEncoder();
> ```
> 
> i switch to version 1.1.1 and my code suddenly not compiling. To make it compile, i have to change the usage to
> 
> ```
> const auto p = gepaf::PolylineEncoder<>();
> ```
> 
> The interface break is not allowed within the same major version, but it happened. Hence the new namespace i made and the old namespace with the using.

Ah, I see now what you mean. You are referring to [this ](https://github.com/vahancho/polylineencoder/releases/tag/1.1.0) change, where the `PolylineEncoder` class had been transformed into a class template. This was, indeed, a major refactoring and the interfaces change. I increased the minor version number only, or even introduced versioning first time.
But I really don't want this `gepaf2` namespace, and the namespace changes at all. What if we change the `major` version number instead, like: 2.0.0?

Reply:
hey, sorry for the late reply!

what i suggest is the following: let this PR go into the codebase, add a tag 1.1.2 (to have the latest version of 1.x.x compatible) and then remove the gepaf2 and make the tag 2.0.0. For me this is the cleanest solution.

Reply:
even i can make the changes if it's okay for you

Reply:
> what i suggest is the following: let this PR go into the codebase, add a tag 1.1.2 (to have the latest version of 1.x.x compatible) and then remove the gepaf2 and make the tag 2.0.0. For me this is the cleanest solution.

Ok, sounds good. Let's do it, but please remove the Cmake related changes. I'm going to make it properly (see the [PR](https://github.com/vahancho/polylineencoder/pull/12)).

Reply:
deal and done